src/java.base/share/classes/java/lang/Module.java

```diff
@@ -597,6 +597,14 @@ public boolean isExported(String pn, Module other) {
      *
      * <p> This method does not check if the given module reads this module. </p>
      *
+     * @apiNote A package {@code p} opened to module {@code M} means that code in
+     * {@code M} can do {@linkplain java.lang.reflect.AccessibleObject#setAccessible(boolean)
+     * deep reflection} on all types in the package.
+     * Further, if {@code M} reads this module, it can obtain a
+     * {@link java.lang.invoke.MethodHandles.Lookup Lookup} object that is allowed to
+     * {@link java.lang.invoke.MethodHandles.Lookup#defineClass(byte[]) define classes}
+     * in package {@code p}.
```

Comment:
A small suggestion here is to change "means that code in M" to "allows code in M".